src/services/kratos/index.ts

```diff
@@ -32,3 +32,25 @@ export const validateKratosToken = async (
     session,
   }
 }
+
+export const validateKratosCookie = async (
+  cookie: string,
+): Promise<ValidateKratosTokenResult | KratosError> => {
+  let session: Session
+
+  try {
+    const { data } = await kratosPublic.toSession({ cookie })
+    session = toDomainSession(data)
+  } catch (err) {
+    if (err.message === "Request failed with status code 401") {
+      return new AuthenticationKratosError(err)
+    }
+    return new UnknownKratosError(err)
+  }
+
+  // TODO: should return aal level also
+  return {
+    kratosUserId: session.identity.id,
+    session,
+  }
+}
```

Comment:
I don't think we need this function

the reason we are parsing the Token when they arrive in the header, is because we have to manage the legacy JWT + new kratos token. so there is some logic in the backend.

for cookie, we can have oathkeeper managing the authentication directly, and returning the user directly through the header (with a signed asymmetric JWT)